app/models/ems_folder.rb

```diff
@@ -166,6 +166,23 @@ def folder_path_objs(*args)
     folders
   end
 
+  # similar to folder_path_objs, but takes in a scope
+  # @param folder [Relation] relation pointin to ems folder of interest
+  def self.folder_path_objs(folder, *args)
+    options = args.extract_options!
+    folder_rels = Relationship.find_by(:resource_id => folder.select(:id), :resource_type => "EmsFolder")
+    if folder_rels.nil? # parent folder has no parents, use the parent folder
```

Comment:
added support for `relationship_types` - good eye. We'll talk when we get back from vaca